src/core/core.controller.js

```diff
@@ -169,8 +169,18 @@ module.exports = function(Chart) {
 
 			// the canvas render width and height will be casted to integers so make sure that
 			// the canvas display style uses the same integer values to avoid blurring effect.
-			var newWidth = Math.floor(helpers.getMaximumWidth(canvas));
-			var newHeight = Math.floor(aspectRatio? newWidth / aspectRatio : helpers.getMaximumHeight(canvas));
+
+            // Default values. Set to 0 instead of canvas.size because the size defaults to 300x150 if the element is collased
+			var newWidth = 0;
+			var newHeight = 0;
+
+            // Avoid issues with the canvas having negative maximum width and/or height due to element being collapsed
+			if (Math.floor(helpers.getMaximumWidth(canvas)) >= 0) {
```

Comment:
I would change just just slightly to cache the return of `helpers.getMaximumWidth(canvas)` into a local variable. This would improve minification and performance. The max width function reads some styles off the canvas element which can be slow